gitfile/commit.go

```diff
@@ -47,10 +49,17 @@ func CommitCreate(d *schema.ResourceData, meta interface{}) error {
 	commit_message := d.Get("commit_message").(string)
 	filepath := parseHandle(handle).path
 
+	if out, err := gitCommand(checkout_dir, "diff", "--cached", "--", filepath); err != nil {
```

Comment:
Possible fix 1: check if there's anything to be committed. Bail out if not. By returning nil, the statefile is not updated